sawtooth_sdk/processor/core.py

```diff
@@ -95,7 +102,8 @@ def _register_requests(self):
                 [TpRegisterRequest(
                     family=n,
                     version=v,
-                    namespaces=h.namespaces)
+                    namespaces=h.namespaces,
+                    protocol_version=SDK_PROTOCOL_VERSION)
```

Comment:
This should only be sent if it is needed, otherwise, a newer SDK could never talk to a 1.1 or older validator.  In other words, the protocol version is only `1` for registrations requiring the version 1 features (i.e. raw headers).

Reply:
A feature - version table needs to be maintained in the SDK. Is that a good practice?

Reply:
I'm not sure if it's a good practice, but we shouldn't have to expose to the developer whether or not they need to send the correct version to the validator.  In other words, this should be handled by the SDK

Reply:
How about maintaining a table/hashmap internally in the SDK, that would have the feature as key and the supported version as its value. If a developer enables these features, SDK would map the feature to the version it is supported.
Concern is that this hashmap/table has to be maintained for long term support.

Reply:
Probably reasonable.  I'll have to see it implemented to be sure I understand what you mean.

Reply:
For reference related discussion of versioning on the validator side: https://github.com/hyperledger/sawtooth-core/pull/1995

Perhaps at a semver major version we clear out any of this handling from the SDK. A v2 SDK only talks to a v2 validator. That would limit some of the maintainability costs of this handling.

Reply:
I don't agree with that.  The only thing that requires this protocol version bump is an opt-in feature.  I don't really see this becoming a long list, frankly.

Reply:
@peterschwarz I attempted to address this concern as you requested.  Syncing @vaporos on this change.